### PR TITLE
rebar3: enable compatibility with multiple Erlang versions

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`rebar3` is designed to work with multiple versions of Erlang. However,
it can only be used with versions of Erlang that are at least as new as
the one it was built with.

To maximise compatibility with supported versions of Erlang, let's build
with `erlang@25`. We also adjust the `install` method to ensure that
we're always building with the oldest supported version of Erlang.

### CODE REVIEW

**Review by nandahkrishna — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
@carlocab, after this I thought of a different solution. `rebar3` already packages pre-compiled Erlang binaries and releases those, per version. You won't have the `./bootstrap` phase, but that should be Ok, because tests will prove it's compatible with multiple arch.s

Ex.: [rebar3 3.24.0 (this is the download link)](https://github.com/erlang/rebar3/releases/download/3.24.0/rebar3)

**Comment by paulo-ferraz-oliveira** [comment]
Yeah, on the other hand you still need Erlang as a runtime dep., so maybe what we have should be enough. I'll look at it closer...

File: Formula/r/rebar3.rb [hunk 1]
```diff
@@ -12,18 +12,32 @@ class Rebar3 < Formula
...
+    rebuild 2
+    sha256 cellar: :any_skip_relocation, arm64_sequoia: "6c00aba8ef8aa1954959241927a697dbb389af2e53113013ec740950859134b7"
+    sha256 cellar: :any_skip_relocation, arm64_sonoma:  "081a82ce85c737630c4f454e32031c095b5f7fb8c9139705c26ea1463b56bab2"
+    sha256 cellar: :any_skip_relocation, arm64_ventura: "daaf2e1860ada7bcaff3c59dd161611f4af4298893ee19d6b543523eecc060fd"
+    sha256 cellar: :any_skip_relocation, sonoma:        "c53cf3cf89d153a1c22c75077eb23e917806af58aac6ab58053537227a438970"
+    sha256 cellar: :any_skip_relocation, ventura:       "96c91363e3a00fc8b4cf1187601941bda4dff426e3e79f06f924225ea6de1c2f"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:  "33ffd13dc4476f1006a2b10b00fbdc818178f902299d0384a98362c929268954"
   end
 
+  depends_on "erlang@25" => [:build, :test]
```
Comment (paulo-ferraz-oliveira) [inline Formula/r/rebar3.rb:24]: When Erlang/OTP 28 gets out will this be auto-bumped to 26?

Comment (paulo-ferraz-oliveira) [reply Formula/r/rebar3.rb:24]: Or maybe the test below is enough to know something need to be updated. Yeah, once a year is not that bad (that the Erlang major cycle)

Comment (carlocab) [reply Formula/r/rebar3.rb:24]: > When Erlang/OTP 28 gets out will this be auto-bumped to 26?

Sadly no. We could make it something like
```ruby
depends_on "erlang@#{wanted_erlang_version}" => [:build, :test]
```
but then that doesn't work very well when we generate the package metadata that most `brew` users consume when they `brew install` things. (Most users won't have a copy of this repository checked out anymore.)

> Or maybe the test below is enough to know something need to be updated. Yeah, once a year is not that bad (that the Erlang major cycle)

Yes, that's the idea.

Comment (paulo-ferraz-oliveira) [reply Formula/r/rebar3.rb:24]: I'll try to cook something up, if time allows, where I set `erlang` as a runtime dependency, which it is, but download the resources from GitHub directly (pre-compiled).

Comment (carlocab) [reply Formula/r/rebar3.rb:24]: > I'll try to cook something up, if time allows, where I set `erlang` as a runtime dependency, which it is, but download the resources from GitHub directly (pre-compiled).

Thanks for the offer, but we don't want to install pre-compiled stuff: https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

Comment (paulo-ferraz-oliveira) [reply Formula/r/rebar3.rb:24]: Got it!

**Pull request merged**